### PR TITLE
CAPSTONE: Form_Validaton

### PR DESCRIPTION
**Pull request by paulshaun-kdtk**
## Changes

- added email rule to restrict user from submitting email with uppercase
- added error messages

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+const showError = () => {
+  if (email.value !== email.value.toLowerCase()) {
+    error.textContent = 'Email address must be in lower case';
+  }
+};
+
+email.addEventListener('input', () => {
+  if (email.value === email.value.toLowerCase()) {
+    error.textContent = '';
```
Comment (ridwanediallo) [inline index.js:192]: - [ ] Excellent work on the validation 👏. However, when we try to send an email with uppercase characters, we do not receive the error message. We simply cannot submit the form if the email does not include an uppercase letter. Please make sure that when the email contains only lowercase letters, we can submit the form. As stated in the requirements, if it contains even one uppercase letter, we should see an error message below the submit button. This is going to give a better user experience to your project. 👌 😊

### Demo :point_down: 
[Screencast from 23.06.2023 23:37:40.webm](https://github.com/paulshaun-kdtk/Microverse_Proffessional_Portifolio/assets/89228743/99909048-bf33-494c-b593-cfd173c0c56d)

#### Requirement 
Screenshot from 2023-06-23 23-50-53

**Review by ridwanediallo — CHANGES_REQUESTED** [review]
# Changes Required ♻️
## Hi  @paulshaun-kdtk , 
AlmostThereReachGIF

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow  correctly. ✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.js (hunk 1 above)
Comment (usorfaitheloho) [reply index.js:192]: Kindly implement the changes suggested by the previous reviewer, as your form works perfectly on the mobile view, however on the desktop view, it doesn't submit at all and doesn't check for validation. It is highly important that your project follows the project requirement 😄 


https://github.com/paulshaun-kdtk/Microverse_Proffessional_Portifolio/assets/20537712/d6bdd374-e5c3-49f6-add6-bb017b61f04b

**Review by usorfaitheloho — CHANGES_REQUESTED** [review]
> Hi @paulshaun-kdtk  ,
> 
> Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

  [ ](https://camo.githubusercontent.com/679757134a8fee76cfe1941f9f3ecb0e0cf56dd02da4498268ce31f0dacdfca3/68747470733a2f2f6d65646961302e67697068792e636f6d2f6d656469612f556461535559436c723932636a57706556782f67697068792e676966)
> 
> ## Highlights
> * All linter checks passed ✔️
> * Descriptive pull request ✔️
> * Used GitHub flow ✔️
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html [hunk 2]
```diff
@@ -1,372 +1,216 @@
...
-</html>
+        </fieldset>
+        <input class="name" type="text" name="full-name" id="full-name" placeholder="Full name" maxlength="30"
+          required />
+        <input class="email" type="email" name="email" id="email" placeholder="Email Address" required />
+        <textarea class="comment" name="comment" id="comment" cols="30" rows="5" maxlength="500"
+          placeholder="Write something in me..." required></textarea>
+        <span class="error"></span>
+        <input class="btn form-button" type="submit" value="Get in touch" /><br />
+      </form>
```
Comment (V-Blaze) [inline index.html:189]: - [ ] You have done great so far implementing some of the changes requested by the previous reviewer, :+1: But I think there are some changes you still need to make before moving to the next milestone. :+1: 
- [ ] As pointed out by the previous reviewer, after filling in an invalid email, the error message doesn't show. If you look at the video below, you can see an error in the console. There seems to be an issue with your form input elements. I'll suggest you go through your code and fix this. :smile: :airplane: 

- - VIDEO REFERENCE:

https://github.com/paulshaun-kdtk/Microverse_Proffessional_Portifolio/assets/107632780/41d50570-067e-4341-b921-2deffe330b0e

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @paulshaun-kdtk , 

# Almost there 🟡 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, and some changes from the previous reviewers, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Correct Github flow✔

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- [OPTIONAL] This was supposed to be pointed out by the first reviewer. Your pull request doesn't have the complete linters setup. As part of the project requirement, you need to set up both `stylint` and also `eslint` javascript linters, I'll suggest visiting the [Microverse Linters repository](https://github.com/microverseinc/linters-config/tree/master/html-css-js) and following the directions. make sure to also update your `linters.yml` file.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by paulshaun-kdtk** [comment]
HI @V-Blaze are you sure about what you are saying because when I'm checking it it's doing exactly what you're saying

**Pull request merged**